ui/tests/integration/components/filter-input-explicit-test.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, typeIn, click } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { GENERAL } from 'vault/tests/helpers/general-selectors';
+import sinon from 'sinon';
+
+module('Integration | Component | filter-input-explicit', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.handleSearch = sinon.spy();
+    this.handleInput = sinon.spy();
+    this.handleKeyDown = sinon.spy();
+    this.query = '';
+    this.placeholder = 'Filter roles';
+
+    this.renderComponent = () => {
+      return render(
+        hbs`<FilterInputExplicit aria-label="test-component" @placeholder={{this.placeholder}} @query={{this.query}} @handleSearch={{this.handleSearch}} @handleInput={{this.handleInput}} @handleKeyDown={{this.handleKeyDown}} />`
+      );
+    };
+  });
+
+  test('it renders', async function (assert) {
+    assert.expect(2);
+    this.query = 'foo';
+    await this.renderComponent();
+
+    assert
+      .dom(GENERAL.filterInputExplicit)
+      .hasAttribute('placeholder', 'Filter roles', 'Placeholder passed to input element');
+    assert.dom(GENERAL.filterInputExplicit).hasValue('foo', 'Value passed to input element');
+  });
+
+  test('it should call handleSearch on submit', async function (assert) {
+    assert.expect(1);
+
+    this.handleSearch = () => {
+      assert.ok(true, 'handleSearch was called');
+    };
```

Comment:
```suggestion
      assert.true(this.handleSearch.calledWith('bar'), 'handleSearch was called');
```

we don't need to explicitly define `this.handleSearch` again down here since you've already set it up with a sinon at the top level `beforeEach`.

once that change is made, you can simply check for things like `this.handleSearch.called`, `this.handleSearch.calledWith('xyz')`. `this.handleSearch.calledOnce` or [any of the other sinon.spy methods](https://sinonjs.org/releases/latest/spies/) in any of the tests down here. 

that's the nice thing about sinon -- it gives you all kinds of utility functions to assert on, so there's no need to use `assert.ok(true)` 🎉